mne_bids/path.py

```diff
@@ -762,31 +840,47 @@ def _find_best_candidates(params, candidate_list):
     return best_candidates
 
 
-def _get_bids_fname_from_filesystem(*, bids_basename, bids_root, sub, ses,
-                                    kind):
+def _get_bids_fpath_from_filesystem(*, bids_basename, bids_root, sub, ses,
+                                    kind, extension):
     if kind is None:
         kind = _infer_kind(bids_basename=bids_basename, bids_root=bids_root,
                            sub=sub, ses=ses)
 
-    data_dir = make_bids_folders(subject=sub, session=ses, kind=kind,
-                                 make_dir=False)
+    # support files above the bids_root/**/<kind>/ level
+    if kind == 'scans':
+        data_dir = make_bids_folders(subject=sub, session=ses,
+                                     make_dir=False)
+    else:
+        data_dir = make_bids_folders(subject=sub, session=ses, kind=kind,
+                                     make_dir=False)
 
-    bti_dir = op.join(bids_root, data_dir, f'{bids_basename}_{kind}')
+    bti_dir = op.join(bids_root, data_dir, f'{bids_basename}')
     if op.isdir(bti_dir):
         logger.info(f'Assuming BTi data in {bti_dir}')
-        bids_fname = f'{bti_dir}.pdf'
+        bids_fpath = f'{bti_dir}.pdf'
     else:
         # Find all matching files in all supported formats.
-        valid_exts = list(reader.keys())
-        matching_paths = glob.glob(op.join(bids_root, data_dir,
-                                           f'{bids_basename}_{kind}.*'))
+        valid_exts = ALLOWED_FILENAME_EXTENSIONS
+        search_str = op.join(bids_root, data_dir,
+                             f'{bids_basename}*')
+        matching_paths = glob.glob(search_str)
         matching_paths = [p for p in matching_paths
                           if _parse_ext(p)[1] in valid_exts]
 
+        if len(matching_paths) > 1:
+            # only get the raw (m/i)EEG dataset if more then one match
```

Comment:
(m/i)EEG -> MEG/EEG/iEEG

I stared at this for too long

Reply:
LOL my bad :p.